File: Backend-POS/POS.Main/POS.Main.Business.Menu/Models/OptionGroup/UpdateOptionGroupRequestModel.cs
using System.ComponentModel.DataAnnotations;

namespace POS.Main.Business.Menu.Models.OptionGroup;

public class UpdateOptionGroupRequestModel
{
    [Required(ErrorMessage = "กรุณาระบุชื่อกลุ่มตัวเลือก")]
    [StringLength(100)]
    public string Name { get; set; } = string.Empty;

    public bool IsRequired { get; set; } = false;
    public int MinSelect { get; set; } = 0;
    public int? MaxSelect { get; set; }
    public bool IsActive { get; set; } = true;

    [Required(ErrorMessage = "ต้องมีตัวเลือกอย่างน้อย 1 รายการ")]
    [MinLength(1, ErrorMessage = "ต้องมีตัวเลือกอย่างน้อย 1 รายการ")]
    public List<OptionItemRequestModel> OptionItems { get; set; } = new();
}

File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/update-option-group-request-model.ts
/* tslint:disable */
/* eslint-disable */
import { OptionItemRequestModel } from '../models/option-item-request-model';
export interface UpdateOptionGroupRequestModel {
  isActive?: boolean;
  isRequired?: boolean;
  maxSelect?: number | null;
  minSelect?: number;
  name: string;
  optionItems: Array<OptionItemRequestModel>;
}
